### ISSUE TITLE
Model error or code bug?

### ISSUE THREAD

**Issue by tomimaki** [scp-fs2open/fs2open.github.com#1080]
In BP AoA mission "The Great Preservers" (bp-15.fs2) I got crash on debug build with this: `ASSERTION: "tswp->current_primary_bank < tswp->num_primary_banks" at aicode.cpp:8756`
It happens during attempt destroy Abalam. Someone can say this is model error, but I don't get crash on nightly 12 August 2016 or earlier. I can't check few later builds bcs of other bugs that were fixed, but on 20 august or later nightly I see this assertion. I suspect bitset flags conversion but can't be sure...

backtrace:
```
Thread 1 "fs2_open_3_7_5_" received signal SIGTRAP, Trace/breakpoint trap.
debug_int3 (file=0x555555f89938 "/home/tomi/fs2open.github.com/code/osapi/dialogs.cpp", line=350) at /home/tomi/fs2open.github.com/code/osapi/osapi.cpp:378
378		gr_activate(1);
(gdb) bt
#0  debug_int3 (file=0x555555f89938 "/home/tomi/fs2open.github.com/code/osapi/dialogs.cpp", line=350) at /home/tomi/fs2open.github.com/code/osapi/osapi.cpp:378
#1  0x0000555555cc07a4 in os::dialogs::Error (text=0x555574efe3b0 "Assert: \"tswp->current_primary_bank < tswp->num_primary_banks\"\nFile: aicode.cpp\nLine: 8756\n\n  /home/tomi/freespace2/fs2_open_3_7_5_x64-DEBUG : os::dialogs::AssertMessage(char const*, char const*, int,"...) at /home/tomi/fs2open.github.com/code/osapi/dialogs.cpp:350
#2  0x0000555555cbf94c in os::dialogs::AssertMessage (text=0x555555f39078 "tswp->current_primary_bank < tswp->num_primary_banks", filename=0x555555f3731e "aicode.cpp", linenum=8756, format=0x0) at /home/tomi/fs2open.github.com/code/osapi/dialogs.cpp:143
#3  0x000055555591a271 in ai_chase () at /home/tomi/fs2open.github.com/code/ai/aicode.cpp:8756
#4  0x0000555555927502 in ai_execute_behavior (aip=0x55555639ffa0 <Ai_info+34400>) at /home/tomi/fs2open.github.com/code/ai/aicode.cpp:13031
#5  0x000055555592a869 in ai_frame (objnum=2452) at /home/tomi/fs2open.github.com/code/ai/aicode.cpp:14166
#6  0x000055555592aca6 in ai_process (obj=0x5555569f6a00 <Objects+1373120>, ai_index=25, frametime=0.0247955322) at /home/tomi/fs2open.github.com/code/ai/aicode.cpp:14252
#7  0x0000555555df5a2d in ship_process_post (obj=0x5555569f6a00 <Objects+1373120>, frametime=0.0247955322) at /home/tomi/fs2open.github.com/code/ship/ship.cpp:8945
#8  0x0000555555caa93a in obj_move_all_post (objp=0x5555569f6a00 <Objects+1373120>, frametime=0.0247955322) at /home/tomi/fs2open.github.com/code/object/object.cpp:1233
#9  0x0000555555cab65d in obj_move_all (frametime=0.0247955322) at /home/tomi/fs2open.github.com/code/object/object.cpp:1457
#10 0x00005555558f24dc in game_simulation_frame () at /home/tomi/fs2open.github.com/freespace2/freespace.cpp:3869
#11 0x00005555558f30f2 in game_frame (paused=false) at /home/tomi/fs2open.github.com/freespace2/freespace.cpp:4262
#12 0x00005555558f3df2 in game_do_frame () at /home/tomi/fs2open.github.com/freespace2/freespace.cpp:4671
#13 0x00005555558f624e in game_do_state (state=2) at /home/tomi/fs2open.github.com/freespace2/freespace.cpp:6368
#14 0x00005555559ccb69 in gameseq_process_events () at /home/tomi/fs2open.github.com/code/gamesequence/gamesequence.cpp:409
#15 0x00005555558f6f77 in game_main (argc=1, argv=0x7fffffffe278) at /home/tomi/fs2open.github.com/freespace2/freespace.cpp:6904
#16 0x00005555558f93af in actual_main (argc=1, argv=0x7fffffffe278) at /home/tomi/fs2open.github.com/freespace2/freespace.cpp:8109
#17 0x00005555558f952c in main (argc=1, argv=0x7fffffffe278) at /home/tomi/fs2open.github.com/freespace2/freespace.cpp:8154
```

**Comment by DahBlount** [comment]
Issue with this being a bug in the code is that the section of code involved hasn't been changed since 2008 according to git blame. If this is occurring, could you also get the values associated tswp->current_primary_bank and tswp->num_primary_banks at the time of the assertion alongside the stack trace?

**Comment by MageKing17** [comment]
> Issue with this being a bug in the code is that the section of code involved hasn't been changed since 2008 according to git blame.

How recently the code containing the assertion that tripped has changed is irrelevant, because the problem obviously occurred elsewhere when `current_primary_bank` became an invalid value.

This bug was ringing an annoying bell in the back of my brain; I'm positive I had tracked down a similar issue earlier this year, but I couldn't recall the details. Then I decided to check my IRC logs:

> `Sep 08 14:58:45 <Axem>	ASSERTION: "tswp->current_primary_bank < tswp->num_primary_banks" at aicode.cpp:8756`
> `Sep 08 14:58:49 <Axem>	so this is what im tripping`
> `Sep 08 14:59:06 <Axem>	ive got VS attached in debug mode right now`
> `Sep 08 14:59:27 <Axem>	both of those two values are 1 right now`

After some investigation, I pointed Axem towards [Mantis issue 2715](http://scp.indiegames.us/mantis/view.php?id=2715) and, as far as I know, the problem went away when he filled out the ships' loadouts; the problem here is that this is a very *old* bug (reported in 2012), so it doesn't make sense that this assertion would've suddenly started tripping in August.
